packages/atlas/src/views/studio/PlaylistWorkspace/VideoSelectorDialog/VideoSelectorDialog.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC, useState } from 'react'
+
+import { useFullVideosConnection } from '@/api/hooks/videosConnection'
+import { VideoOrderByInput } from '@/api/queries/__generated__/baseTypes.generated'
+import { SvgActionSearch } from '@/assets/icons'
+import { Text } from '@/components/Text'
+import { Button } from '@/components/_buttons/Button'
+import { VideoListItem, VideoListItemLoader } from '@/components/_video/VideoListItem'
+import { useDebounceValue } from '@/hooks/useDebounceValue'
+import { useAuthorizedUser } from '@/providers/user/user.hooks'
+import { createId } from '@/utils/createId'
+import { SentryLogger } from '@/utils/logs'
+
+import { ListWrapper, StyledDialogModal, StyledInput, Wrapper } from './VideoSelectorDialog.styles'
+
+type VideoSelectorDialogProps = {
+  onHide: () => void
+  show: boolean
+  onSelect: (arg: [string, string][]) => void
+}
+const INITIAL_FIRST = 50
+
+export const VideoSelectorDialog: FC<VideoSelectorDialogProps> = ({ onHide, show, onSelect }) => {
+  const [selectedVideoIds, setSelectedVideoIds] = useState<string[]>([])
+  const [search, setSearch] = useState<string>('')
+  const { channelId } = useAuthorizedUser()
+  const debouncedSearch = useDebounceValue(search, 200)
+  const { edges, loading } = useFullVideosConnection(
+    {
+      first: INITIAL_FIRST,
+      orderBy: VideoOrderByInput.CreatedAtDesc,
+      where: {
+        channel: {
+          id_eq: channelId,
+        },
+        title_contains: debouncedSearch,
+      },
+    },
+    {
+      notifyOnNetworkStatusChange: true,
+      onError: (error) => SentryLogger.error('Failed to fetch videos', 'MyVideosView', error),
+    }
+  )
+
+  return (
+    <StyledDialogModal
+      show={show}
+      size="medium"
+      contentClassName="dialog-content"
+      primaryButton={{
+        text: 'Add videos',
+        disabled: !selectedVideoIds.length,
+        onClick: () => {
+          onSelect(selectedVideoIds.map((videoId) => [videoId, createId()]))
+          setSelectedVideoIds([])
+          onHide()
+        },
+      }}
+      secondaryButton={{ text: 'Cancel', onClick: onHide }}
```

Comment:
once someone clicks cancel I think we should:
- unselect all the videos
- clear the search

Right now you can click cancel and open the dialog again and you'll still see the same videos checked, which I don't think is a good user experience